### PR TITLE
lib-user: refactor group stats page header items and content per group, user, and membership

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Describe your changes
- add header link to MyGroups
- refactor GroupStats header items per user, group, and membership (role)
- add initial leave group (delete membership) functionality
- refactor GroupStats TopContributors per user, group, and membership (role)

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - review different group types as different roles
  - you'll likely want to create a group to test as group_admin, but here are some staging groups to test as a group_member (app-root `yarn dev`):
    - public aggregate only: https://local.zooniverse.org:3000/groups/1326971?join_token=a7dd6ca55dc96f1d
    - public member stats: https://local.zooniverse.org:3000/groups/1326991?join_token=9a7853ccece01473
    - public show all stats: https://local.zooniverse.org:3000/groups/1326989?join_token=3df585676014b6b8
    - private aggregate only: https://local.zooniverse.org:3000/groups/1326984?join_token=62403a6204292768
    - private member stats: https://local.zooniverse.org:3000/groups/1326986?join_token=a1428f952269778b
- Which storybook stories should be reviewed?
  - TopContributors:
    - http://localhost:6008/?path=/story/components-groupstats-topcontributors--default
    - http://localhost:6008/?path=/story/components-shared-contentbox--top-contributors
- Are there plans for follow up PR’s to further fix this bug or develop this feature? see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [x] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Review by goplayoutside3 — COMMENTED** [review]
I was able to join a public group at: https://local.zooniverse.org:3000/groups/1326991
I made contributions and left the group 👍 Once I left the group, the component refreshed and my account was removed from the contributors list.

I was a member of group https://local.zooniverse.org:3000/groups/1326984, which is a _private_, agg only group. When I left the group while viewing that url, the component did not refresh until I manually refreshed the page though, so I could still see the private group even though I was no longer a member. Just want to check if you're seeing the same behavior for private groups? If you need one of my private groups to test, here's a staging link: https://www.zooniverse.org/groups/1326995?join_token=0e834515f7390aa4

(Also note the group links in your PR description are all for production and would be easier to click if linked to app-root run locally).

**Comment by mcbouslog** [comment]
Yes I saw the same behavior after leaving. I added a refactor route a user to their profile page after if the membership delete is successful. I had an iteration that stayed on the group stats page if it was a public group, but thought that might be confusing? I could add that back, as I could see an argument accordingly (stay on group stats page if public, route elsewhere if private). I think there are a few other similar situations I can bring up in related user_group meeting.

File: packages/app-root/src/app/groups/[groupId]/GroupStatsContainer.js [hunk 1]
```diff
@@ -9,7 +9,13 @@ function GroupStatsContainer({
   groupId,
   joinToken
 }) {
-  const { adminMode, user } = useContext(PanoptesAuthContext)
+  const { adminMode, isLoading, user } = useContext(PanoptesAuthContext)
+
+  if (typeof window === 'undefined' || isLoading) {
+    return (
+      <p>Loading...</p>
```
Comment (goplayoutside3) [inline packages/app-root/src/app/groups/[groupId]/GroupStatsContainer.js:16]: Just a side note that you can now use `<Loader/>` from lib-react-components for any loading UI if desired!

File: packages/lib-user/src/components/GroupStats/DeleteGroup.js [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+  deletePanoptesUserGroup
+} from '@utils'
 
-    if (confirmDelete) {
-      handleGroupDelete()
+function DeleteGroup({ groupId }) {
+  async function handleGroupDelete({ groupId }) {
+    try {
+      const deleteResponse = await deletePanoptesUserGroup({ groupId })
+      console.log('deleteResponse', deleteResponse)
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/GroupStats/DeleteGroup.js:11]: Intentionally left in?

File: packages/lib-user/src/components/GroupStats/EditGroup.js [hunk 3]
```diff
@@ -13,6 +17,16 @@ function EditGroup({ group, handleGroupUpdate }) {
     setStatsVisibility(group?.stats_visibility || '')
   }, [group])
 
+  async function handleGroupUpdate({ groupId, data }) {
+    try {
+      const updatedGroup = await updatePanoptesUserGroup({ groupId, data })
+      console.log('updatedGroup', updatedGroup)
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/GroupStats/EditGroup.js:23]: Intentionally left in?

File: packages/lib-user/src/components/shared/Layout/Layout.js [hunk 4]
```diff
@@ -174,6 +175,7 @@ function Layout({
           dark: 'dark-3',
           light: 'neutral-6'
         }}
+        forwardedAs='main'
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/Layout/Layout.js:178]: 👍

**Review by goplayoutside3 — APPROVED** [review]
Much better 🎉 When I join and then leave a private or public group, I get redirected.

1. Your comment above brings up a point I want to discuss too. Because a user's profile page is not part of the launch this summer (even though there is a matching new design), I don't think we should link users to it. I think when a user is viewing `/groups/[groupId]` the header "back" button should be to `/users/[login]/groups`, aka My Groups. I don't think it makes sense to send a user "back" to PFE from any of the Groups related pages. For user stats pages, whenever a user is viewing `/users/[login]/stats` I think the "back" button should be to Zooniverse homepage since they're signed in. Let's definitely chat at the next user groups meeting!

    Edit: Expanding on this comment ^ to add that when a user leaves a group, they should be redirected to the My Groups page, too.

2. To double check my understanding of top contributors - the classification count used for a group's contributors only applies when a contributor made the classifications while a member of the group. Is that correct? For instance, my staging account shows I made 7 classifications on Pizza Please via the profile ribbon, but as a member of group 1326986 I made 3 classifications, and therefore my username says "3 classifications" in top contributors of the group page.

**Comment by mcbouslog** [comment]
1. Agreed and sounds good about revisiting how users are routed after certain group actions

2. That's correct, user classifications only count towards group classifications while the user is a member of the group

**Comment by mcbouslog** [comment]
Noting these code changes include console logging the resource responses after an action (i.e. deleting a group), but I've included iterations on that in this subsequent PR (i.e. f8009e884a8bc7e7c04674547928444787d306aa).

**Pull request merged**